src/routes/jira/jira-get-connected-repos.ts

```diff
@@ -0,0 +1,139 @@
+import { NextFunction, Request, Response } from "express";
+import { RepoSyncState } from "~/src/models/reposyncstate";
+import { Subscription, TaskStatus } from "~/src/models/subscription";
+
+interface Page {
+	pageNum: number;
+	isCurrentPage: boolean;
+}
+
+export const JiraGetConnectedRepos = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+): Promise<void> => {
+
+	try {
+		const { jiraHost, nonce } = res.locals;
+		const subscriptionId = Number(req.params.subscriptionId) || Number(req.body.subscriptionId);
+		const page = Number(req.query.page || 1);
+
+		if (!jiraHost) {
+			req.log.warn({ jiraHost, req, res }, "Missing jiraHost");
+			res.status(404).send(`Missing Jira Host`);
+			return;
+		}
+
+		if (!subscriptionId) {
+			req.log.error("Missing Subscription ID");
+			res.status(401).send("Missing Subscription ID");
+			return;
+		}
+
+		const subscription = await Subscription.findByPk(subscriptionId);
+
+		const repoSyncStates = await RepoSyncState.findAllFromSubscription(subscription);
```

Comment:
May something like `await RepoSyncState.findAllFromSubscription(subscription, { order: [["id", "DESC"]] });`?
As otherwise the pagination may not be stable?

Reply:
I've added ordering to the findAllFromSubscription function itself